Solve warnings and remove deprecated code

bumbummen99 opened the pull request:
As the title says, this Pull Request will solve thrown Warnings and remove deprecated code.

Close #726

bumbummen99 committed:
[Fix] PHPDoc annotations for improved static analysis,
[Change] Drop use of soon to be removed utf8_encode / utf8_decode functions,
[Change] Prevent sprintf template being null on getJoinLink(),
[Fix] Protocols::joinLink default value and type hints,

bumbummen99 committed:
[Fix] Deprecated pass preg_replace subject as null,
[Change] Refactor Stripcolors filter implementation

bumbummen99 committed:
[Fix] Implicit conversion losing precision from float to integer

bumbummen99 committed:
[Fix] Deprecated dynamic property in test Issue588

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/63ee084d-4649-40f9-b25b-524ca0b51a2d) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/63ee084d-4649-40f9-b25b-524ca0b51a2d) completed: **1 new issues, 6 updated code elements**

bumbummen99 committed:
[Add] Array helper Concern and function to use RecursiveArrayIterator in simple manner,
[Change] Refactor Iterators in Filters with applyRecursively,
[Change] Refactor Normalize Filter

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/f8f7355d-5446-4085-87a0-c595312895cc) was created.

bumbummen99 committed:
Codestyle

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/448620af-4ded-45cc-b9e8-0428e5905a2d) was created.

bumbummen99 committed:
Codestyle and remove debug code

bumbummen99 committed:
Codestyle

bumbummen99 committed:
[Fix] Remove non backwards compatible code

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/3cb29f6b-72c0-4aaa-93ed-4c69cef71984) was created.

bumbummen99 committed:
[Fix] Typo

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/d865a992-be31-4807-81b6-284a81aeb4d3) was created.

bumbummen99 committed:
[Fix] Remove return type hint

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/bd7756d9-9096-41c6-8acc-69adab2fe409) was created.

bumbummen99 committed:
[Fix] Remove null coalesce operator

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/7d396d21-a2ee-4078-a32c-79d3ac8875aa) was created.

bumbummen99 committed:
[Fix] check array key before evaluating boolean value

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/4845a7de-cacc-489c-b6ab-88a97f8e66ef) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/f8f7355d-5446-4085-87a0-c595312895cc) completed: **3 new issues, 9 updated code elements**

bumbummen99 committed:
[Change] Remove unused variable

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/c2d0444f-d3ce-4fdd-893f-b822c0730fe7) was created.

bumbummen99 committed:
[Fix] Remove function parameter type hint

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/1f66766c-70a7-4e2d-b5d2-5b734b0b8555) was created.

bumbummen99 committed:
[Fix] Remove further scalar type hints

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/9fee39de-ed47-4294-9e15-aef795787725) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/4845a7de-cacc-489c-b6ab-88a97f8e66ef) completed: **5 new issues, 8 updated code elements**

bumbummen99 committed:
[Change] Add comments to know what is to be deprecated,
[Fix] Convert numbers to string with Stripcolors Filter

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/350cefd1-5266-477a-97a9-61e97f29f3eb) was created.

bumbummen99 committed:
Codestyle

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/af015720-210d-4004-8768-4e8f344114d3) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/9fee39de-ed47-4294-9e15-aef795787725) completed: **4 new issues, 8 updated code elements**

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/af015720-210d-4004-8768-4e8f344114d3) completed: **4 new issues, 7 updated code elements**

bumbummen99 committed:
[Change] Use dedicated helper classes over Traits to prevent polluting classes,
[Change] Manually write back sub Iterator for ArrayIterator in PHP<8.1,
[Change] Refactor the CI workflow matrix,
[Add] Tool act to the DevContainer for local GitHub Workflows,
[Add] Docker in Docker to DevContainer to run act

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/2c456ef3-8f48-4dd2-8411-aeb2ed8e7d18) was created.

bumbummen99 committed:
Codestyle

bumbummen99 committed:
Codestyle

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/47e4ee1e-a2ab-4fb3-8954-20680cf53275) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/2c456ef3-8f48-4dd2-8411-aeb2ed8e7d18) completed: **5 new issues, 10 updated code elements**

bumbummen99 committed:
[Fix] keep build logs directory

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/2a2c2b0f-3462-4076-98f1-9911ef3f28be) was created.

bumbummen99 committed:
[Fix] Handle different way to access array reference,
[Change] Move protected helper helpers into trait,

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/0322aad5-8304-4365-9e69-65acba8a6290) was created.

bumbummen99 committed:
Fix missing static

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/9400eabe-02f8-4ba1-9ad3-b276d08db430) was created.

bumbummen99 committed:
[Change] Protocol packetResponse parameter default value to empty array instead of null,
Close #729

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/29395508-76df-4060-9d0f-971204ccdbd3) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/47e4ee1e-a2ab-4fb3-8954-20680cf53275) completed: **5 new issues, 10 updated code elements**

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/29395508-76df-4060-9d0f-971204ccdbd3) completed: **5 new issues, 12 updated code elements**

bumbummen99 committed:
[Fix] PHPDoc throws for all Protocols

bumbummen99 committed:
[Fix] Source protocol skip non´challenge response

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/995b946c-5b54-4b83-8b6e-a274a714e43d) was created.

bumbummen99 committed:
Codestyle

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/f4a65c25-3bf8-4606-ad13-9cbc5718c3f7) was created.

bumbummen99 committed:
Codestyle

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/ee6b77e2-e29f-4453-aba7-ac025e593969) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/1581761d-480c-43fb-9742-710686318e6f) completed: **5 new issues, 12 updated code elements**

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/ee6b77e2-e29f-4453-aba7-ac025e593969) completed: **5 new issues, 12 updated code elements**

bumbummen99 committed:
[Change] Install act for all users in DevContainer

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/3bcdb6e8-f93f-4950-9c03-8698a89ecdaa) was created.

bumbummen99 committed:
[Change] Use Coveralis action and enable parallel support

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/e5e430f4-a9b4-4e3b-a943-7106bf59dac5) was created.

bumbummen99 committed:
[Remove] Duplicate step in workflow

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/95894225-e566-4786-b8f9-5027c6276935) was created.

bumbummen99 committed:
[Change] Add clover path to Coveralis action

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/3edc6110-9d98-4959-b286-5e95d743868d) was created.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/3bcdb6e8-f93f-4950-9c03-8698a89ecdaa) completed: **5 new issues, 12 updated code elements**

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/Austinb/GameQ/inspections/3edc6110-9d98-4959-b286-5e95d743868d) completed: **5 new issues, 12 updated code elements**
